digital_land/package/dataset.py

```diff
@@ -185,14 +185,48 @@ def add_counts(self):
         self.commit()
         self.disconnect()
 
+    def entry_date_upsert(self, table, fields, row, conflict_fields, update_fields):
+        """
+        Dataset specific upsert function that only replace values for more recent entry_dates.
+        Will insert rows where no conflict is found. where there's a conflict it was compare entry dates
```

Comment:
It will insert rows where no conflict is found. If there's a conflict it will compare entry dates and 
update to the more recent.